lib/Optimizer/IROptimizer.cpp

```diff
@@ -1366,9 +1366,15 @@ void optimizeInserts(IRFunction &M) {
 
     // Create a new TensorView of the original dest of the InsertTensor,
     // with the same offset as the InsertTensor.
+    std::vector<size_t> tvOffsets(insertDest->dims().size(), 0);
```

Comment:
So it seems this change is because `InsertTensorInsts` do not have their offsets set to the correct number of dimensions as well -- is that correct? So while this change seems sane, it also feels like it might be a bandaid instead of fixing the underlying issue. I.e. would it make sense to also add the same `verify()` check to `InsertTensorInst::verify()` and `ExtractTensorInst::verify()`, to root out the origin of mismatch on offsets/starts?

Reply:
I think so - done.